src/priority_enumerator.jl

```diff
@@ -2,28 +2,58 @@
 Enumerates a context-free grammar up to a given depth in a breadth-first order.
 This means that smaller programs are returned before larger programs.
 """
-mutable struct ContextFreeBFSEnumerator <: ExpressionIterator 
+mutable struct ContextFreePriorityEnumerator <: ExpressionIterator 
     grammar::ContextFreeGrammar
     max_depth::Int
+    # Assigns a priority to a (partial or complete) tree.
+    priority_function::Function
+    # Expands a partial tree.
+    expand_function::Function
     sym::Symbol
+end 
+
+bfs_priority_function(tree::RuleNode, parent_value::Number) = parent_value + 1
+bfs_expand_heuristic(rules) = rules
+
+
+"""
```

Comment:
how about we put these handy constructors in `Utils.jl` in `Search.jl`?
Maybe makes the code a bit clearer to read